api/base.go

```diff
@@ -159,6 +159,13 @@ func (h *BaseHandler) preLoad(db *gorm.DB, fields ...string) (tx *gorm.DB) {
 	return
 }
 
+//
+//
+func (h *BaseHandler) hal(ctx *gin.Context, code int, obj interface{}) {
+	ctx.JSON(code, obj)
+	ctx.Writer.Header().Set("Content-Type", "application/hal+json; charset=utf-8")
```

Comment:
I think there may be a more direct way to set the content-type but can't remember.